test/jdk/javax/security/auth/callback/PasswordCleanup.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2022 THL A29 Limited, a Tencent company. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8284910
+ * @summary Buffer clean in PasswordCallback
+ */
+
+import javax.security.auth.callback.PasswordCallback;
+import java.util.WeakHashMap;
+
+public final class PasswordCleanup {
+    private final static WeakHashMap<PasswordCallback, ?> weakHashMap =
+            new WeakHashMap<>();
+
+    public static void main(String[] args) throws Exception {
+        // Test password clearing at finalization.
+        clearAtCollection();
+
+        // Test password clearing with the specific method.
+        clearWithMethod();
+    }
+
+    private static void clearAtCollection() throws Exception {
+        // Create an object
+        PasswordCallback passwordCallback =
+                new PasswordCallback("Password: ", false);
+        passwordCallback.setPassword("ThisIsAPassword".toCharArray());
+        passwordCallback.clearPassword();
+        weakHashMap.put(passwordCallback, null);
+        passwordCallback = null;
+
+        // Check the clearing
+        checkClearing();
+    }
+
+    private static void clearWithMethod() throws Exception {
+        // Create an object
+        PasswordCallback passwordCallback =
+                new PasswordCallback("Password: ", false);
+        passwordCallback.setPassword("ThisIsAPassword".toCharArray());
+
+        // Use password clear method.
+        passwordCallback.clearPassword();
+
+        weakHashMap.put(passwordCallback, null);
+        passwordCallback = null;
+
+        // Check the clearing
+        checkClearing();
+    }
+
+    private static void checkClearing() throws Exception {
```

Comment:
How is this test testing that the password is cleared?

Reply:
The test case is used to check that the Cleaner used is not bind to 'this' object, and the cleaner during finalization could work.  Unfortunately, as the cleaner behavior is not visible, I don't find a way to automated test that the password is really cleared during finalization.

Reply:
Ok, then I would suggest changing the name of the test as it is misleading. I suggest creating a directory named "PasswordCallback" and then adding a test named perhaps "CheckCleanerNotBoundToThis" or something like that. I would change the name of the `checkClearing` method as you are not checking if passwords are cleared. Also update the @summary to describe what it is actually testing. Use code comments if you need to explain it further.

Reply:
The test has two case now.  One is used to check the PasswordCallback object collection at finalization.  The other one is check the password clearing for clearPassword() method.  Is it better to split into two test files so that the class name could be better fit the purpose?

Reply:
Never mind, I'm splitting the test case into two.